test/python/test_running.py

```diff
@@ -0,0 +1,783 @@
+
+DEFAULT_SCRIPT = 'init.lua'
+DEFAULT_STATEBOARD_SCRIPT = 'stateboard.init.lua'
+
+STATUS_NOT_STARTED = 'NOT STARTED'
+STATUS_RUNNING = 'RUNNING'
+STATUS_STOPPED = 'STOPPED'
+STATUS_FAILED = 'FAILED'
+
+
+# #######
+# Helpers
+# #######
+def write_conf(path, conf):
+    with open(path, 'w') as f:
+        yaml.dump(conf, f, default_flow_style=False)
+
+
+def get_instance_id(app_name, instance_name):
+    return '{}.{}'.format(app_name, instance_name)
+
+
+def get_stateboard_name(app_name):
+    return '{}-stateboard'.format(app_name)
+
+
+class InstanceProcess():
+    def __init__(self, process):
+        self._process = process
+        self._pid = process.pid
+        self._ppid = process.ppid()
+
+        self.name = process.name()
+        self.cmd = process.cmdline()
+
+        env = process.environ()
+
+        assert 'TARANTOOL_APP_NAME' in env
+        if 'TARANTOOL_INSTANCE_NAME' in env:
+            self.id = get_instance_id(
+                env['TARANTOOL_APP_NAME'],
+                env['TARANTOOL_INSTANCE_NAME'],
+            )
+        else:
+            self.id = env['TARANTOOL_APP_NAME']
+
+        self._env = {
+            'TARANTOOL_APP_NAME': env.get('TARANTOOL_APP_NAME'),
+            'TARANTOOL_INSTANCE_NAME': env.get('TARANTOOL_INSTANCE_NAME'),
+            'TARANTOOL_CFG': env.get('TARANTOOL_CFG'),
+            'TARANTOOL_CONSOLE_SOCK': env.get('TARANTOOL_CONSOLE_SOCK'),
+            'TARANTOOL_PID_FILE': env.get('TARANTOOL_PID_FILE'),
+            'TARANTOOL_WORKDIR': env.get('TARANTOOL_WORKDIR'),
+        }
+
+    def is_running(self):
+        return self._process.is_running() and self._process.status() != psutil.STATUS_ZOMBIE
+
+    def getenv(self, name):
+        return self._env.get(name)
+
+
+class Cli():
+    def __init__(self, cartridge_cmd):
+        self._cartridge_cmd = cartridge_cmd
+        self._children = []
+        self._instances = dict()
+
+    def start(self, project, instances=[], daemonized=False, stateboard=False, stateboard_only=False,
+              cfg=None, script=None, run_dir=None, data_dir=None):
+
+        cmd = [self._cartridge_cmd, 'start']
+        if daemonized:
+            cmd.append('-d')
+        if stateboard:
+            cmd.append('--stateboard')
+        if stateboard_only:
+            cmd.append('--stateboard-only')
+        if cfg is not None:
+            cmd.extend(['--cfg', cfg])
+        if script is not None:
+            cmd.extend(['--script', script])
+        if run_dir is not None:
+            cmd.extend(['--run-dir', run_dir])
+        if data_dir is not None:
+            cmd.extend(['--data-dir', data_dir])
+
+        cmd.extend(instances)
+
+        self._subprocess = subprocess.Popen(
+            cmd, cwd=project.path,
+            stdout=sys.stdout,
+            stderr=sys.stderr,
+        )
+
+        self._pid = self._subprocess.pid
+        self._process = psutil.Process(self._pid)
+
+        time.sleep(1)  # let cli to start instances
```

Comment:
I not sure that it's a proper way. Hope it could be changed to "wait_for process is running" or maybe you could ping instance in some way:
```
telnet 0 3301
Trying 0.0.0.0...
Connected to 0.
Escape character is '^]'.
Tarantool 2.3.1 (Binary) 57b40c48-ea17-4dfc-914b-2c371788219d
```